core/src/lib.rs

```diff
@@ -220,14 +186,136 @@ mod tests {
             .entry("birthDate".to_string())
             .or_insert(Value::String("1958-07-17".to_string()));
 
+        return (kv_body, kv_subject);
+    }
+
+    #[test]
+    fn test_create_credential() -> Result<(), String> {
```

Comment:
Guys, Is it possible to create subtests in Rust with closures, how it can be done in GoLang: https://github.com/knox-networks/gomono/pull/570 @mkatychev @LuisOsta ? Thus we could distinct VC creation test from VC creating proof and from Creating Verifiable presentation. Not it is implemented in one test, but maybe we can do subtests like in aforementioned GoLang PR.

Reply:
6t5Why do you need subtests to separate the testing of VCs and VPs? Why not just create a new test like:

```
#[test]
fn test_create_presentation() -> Result<(), String> {
...
}
```
?
And any shared logic can be pulled out into a standard function in the test module?

Reply:
It seems the comments might get duplicated. Yes that's an option and it looks like it is a single option for now, with closure subtest it could be just a more concise way.